### PR TITLE
Feature implement arraySum.js

### PR DESCRIPTION
**Pull request by stillNovice**
- [] Running `yarn lint` does not trigger any linter errors
- [] The test of the method you have fixed is passing
- [] You have written a new skeleton method for someone else to work on!
- [] You have written tests to accompany your skeleton method

### CODE REVIEW

File: lib/arraySum.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-
 function arraySum(array) {
-  // Write code here
+  var sum = 0;
+  Array.prototype.forEach.call(array, (elem) => {
+    if (typeof elem == "number" && !Number.isNaN(elem) && Number.isFinite(elem)) {
+      sum += elem;
+    }
+  });
+  return sum;
```
Comment (ktilcu) [inline lib/arraySum.js]: I'm not sure what the maintainers want but have you tried doing this with `array.reduce` you may find it to be a bit simpler.

Comment (stillNovice) [reply lib/arraySum.js]: @ktilcu thanks for the suggestion. I have used the Array.reduce for brevity.

**Comment by ktilcu** [comment]
Awesome! one last thing. You could return the result of array.reduce directly instead of assigning it to a variable.

**Comment by ktilcu** [comment]
Great! Thanks for being open to feedback @stillNovice! Feel free to pick up another one. Happy Hacktobering!

**Pull request merged**